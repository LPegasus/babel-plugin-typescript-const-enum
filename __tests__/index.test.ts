import * as path from 'path';
import * as fs from 'fs';
import * as babel from '@babel/core';
import babelPluginTypescriptConstEnum from '../src/index';

const fixtures_dirname = path.resolve(__dirname, './fixtures');

it('with renamed namespace work success', async () => {
  const tInput = fs.promises.readFile(
    path.resolve(fixtures_dirname, 'with-renamed-namespace/input.ts'),
    'utf8'
  );
  const input = await tInput;

  const result = await babel.transformAsync(input, {
    presets: ['@babel/preset-typescript'],
    filename: 'input.ts',
    plugins: [
      [
        babelPluginTypescriptConstEnum,
        {
          enums: {
            'OUT_NS_STATUS.success': 0,
            'OUT_NS_STATUS.fail': 1,
            'N.S.STATUS.success': 0,
            'N.S.STATUS.fail': 1,
          },
        },
      ],
    ],
  });

  expect(result?.code).toMatchSnapshot();
});

it('without renamed namespace work success', async () => {
  const tInput = fs.promises.readFile(
    path.resolve(fixtures_dirname, 'without-renamed-namespace/input.ts'),
    'utf8'
  );
  const input = await tInput;

  const result = await babel.transformAsync(input, {
    presets: ['@babel/preset-typescript'],
    filename: 'input.ts',
    plugins: [
      [
        babelPluginTypescriptConstEnum,
        {
          enums: {
            'OUT_NS_STATUS.success': 0,
            'OUT_NS_STATUS.fail': 1,
            'N.S.STATUS.success': 0,
            'N.S.STATUS.fail': 1,
          },
        },
      ],
    ],
  });

  expect(result?.code).toMatchSnapshot();
});

it('with-ts-compiled should be work', async () => {
  const tInput = fs.promises.readFile(
    path.resolve(fixtures_dirname, 'with-ts-compiled/input.js'),
    'utf8'
  );
  const input = await tInput;

  const result = await babel.transformAsync(input, {
    presets: [],
    filename: 'input.js',
    plugins: [
      [
        babelPluginTypescriptConstEnum,
        {
          enums: {
            'HOHO.HAHA_COMMON.ErrorCode.A': 'A',
            'HOHO.HAHA_COMMON.ErrorCode.B': 'B',
            'HOHO.HAHA_COMMON.ErrorCode.C': 'C',
          },
        },
      ],
    ],
  });

  expect(result?.code).toMatchSnapshot();
});
