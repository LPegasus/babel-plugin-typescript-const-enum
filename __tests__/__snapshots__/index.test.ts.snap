// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with renamed namespace work success 1`] = `
"import 'react';
export function foo() {
  const a = 1
  /*fail*/
  ;
  const b = 0
  /*success*/
  ;
  const c = 1
  /*fail*/
  ;
  const d = 0
  /*success*/
  ;

  function baz1() {
    const NS = {
      STATUS: {
        success: 'success',
        fail: 'fail'
      }
    };
    return NS.STATUS.success;
  }

  function baz2() {
    const ONS = {
      success: 'success',
      fail: 'fail'
    };
    return ONS.success;
  }

  function baz3() {
    const N = {
      S: {
        STATUS: {
          success: 'success',
          fail: 'fail'
        }
      }
    };

    function bas3Closure() {
      return N.S.STATUS.fail;
    }

    return bas3Closure();
  }

  function haha() {
    return function () {
      return 0
      /*success*/
      ;
    };
  }

  return;
}"
`;

exports[`with-ts-compiled should be work 1`] = `
"var HOHOHO = {};
var SuccessCode = {
  HOHO: {
    HAHA_COMMON: {
      SuccessCode: {
        OK: 'OK',
        FAIL: 'FAIL'
      }
    }
  }
};
var P = HOHOHO.A;

const ErrorToString = code => {
  const readableString = {
    [\\"A\\"
    /*A*/
    ]: '1',
    [\\"B\\"
    /*B*/
    ]: '2',
    [\\"C\\"
    /*C*/
    ]: '3',
    [SuccessCode.HOHO.HAHA_COMMON.SuccessCode.OK]: '4',
    [SuccessCode.HOHO.HAHA_COMMON.SuccessCode.FAIL]: '5',
    [P.A]: 'undefined'
  }[code];
  return readableString || '未知错误，保留现场，联系研发';
};"
`;

exports[`without renamed namespace work success 1`] = `
"import 'react';
export function foo() {
  console.log(1
  /*fail*/
  );

  function bar() {
    const NS = '';
    return NS;
  }

  bar();
  return;
}"
`;
